### ISSUE TITLE
Multilure and Sonar buff cause item-pickup-texts to be increased by 1

### ISSUE THREAD

**Issue by BarzenX** [bluemagic123/tModLoader#4042]
### Version

Latest 1.4.4-stable

### OS

Windows

### Platform

Steam

### This bug affects...

Gameplay as a Player

### Description

When using multiple fishing lines all works fine until I apply the Sonar buff. After that, the item-pickup-texts from the fished loot get increased by 1 for the "extra spawned" boppers.

### Log File

not needed

### Steps to reproduce

1. Get the MultiLure mod from Adbluescarab (https://steamcommunity.com/sharedfiles/filedetails/?id=2942015919)
2. Configure one of the mods lures to have more than 1 fishing line (I took 10 as example)
3. Create a fresh world and enter with a fresh player
4. Go to a fishing spot (f.ex. the ocean) and equip/buff to have a sufficiently high fishing power to have most of the boppers splashing
5. Reel in the loot with and without the sonar buff
6. Compare the item-pickup-texts item-amount with the actual added amount in the inventory

### Expected Behavior

The pickup texts should always match the added items to the inventory

### Actual Behavior

When the sonar buff is active some pickup-texts state an increase by 1. I assume the original bopper is the one with the correct amount.
Screenshot: https://imgur.com/yntQBIX

### Reproduction frequency

Always

### Additional Information

nothing

**Comment by BarzenX** [comment]
I'm really honored to see that my reported issue really pointed out to something in the tModLoader and that I could take part in improving it 😍
You, Sir, are awesome! 🥰

**Comment by JavidPack** [comment]
Thanks for the report. Turns out there was another bug with the sonar texts not despawning immediately. They should despawn right as the item is given to the player.

Fixed behavior:

https://github.com/tModLoader/tModLoader/assets/4522492/50845be0-553a-41d5-986b-6ac269c295ce

Technically Terraria doesn't support multiple lures, but since it's so common in mods, I went ahead and made a fix for this in tModLoader itself. This fix does technically have the potential to conflict with something some mod will be doing, so we'll make sure it is tested during February for March release.

There is a simpler fix that would only fix the popuptext stack size (and not the despawning), if this fix causes issues with mods during the preview period we can revert to this change in `PopupText.NewText`:

```diff
	if (!popupText.active || popupText.notActuallyAnItem || (!(popupText.name == newItem.AffixName()) && (!flag || !popupText.coinText)) || popupText.NoStack || noStack)
		continue;

+	if (popupText.sonar)
+		continue;
```
